### DISCUSSION TITLE
How to set "exclude type" in input.json

### DISCUSSION THREAD

**Discussion by zhao-w-en** [deepmodeling/dpdata#166] [Q&A]
Can I exclue atom types from a system when training dp model? Cause we think the calculated forces of an atom type maybe not accurate when we fixed them during AIMD. 
       I've read your annotation of "exclude type" in documentation but when I set "exclude type": ["C"], the program goes wrong.

**Comment by njzjz** [comment]
`[[1,2],[2,3]]` means excluding forces between type 1 and type2, and between type2 and type3. Note that the role of forces is mutual.

**Reply by zhao-w-en** [reply]
Will it influence energy fitting if I excluding force between type 1 and type 1? It seems forces fitting are worse and rmse of forces becomes more larger...

**Reply by njzjz** [reply]
Yes, the force is a negative gradient of energy. In fact, when implementing this feature, we only modify the energy code, and the force is calculated automatically.
![image](https://user-images.githubusercontent.com/9496702/125130683-66a6ed80-e0cf-11eb-815c-b6ce8de62abc.png)

Can you confirm both your energy and forces exclude interaction between type 1 and type 1?

**Reply by zhao-w-en** [reply]
In fact I don't quite understand the last sentence. I set "exclude type":[[1,1]] and just as you said the energy and forces between type 1 and type 1 will be exclude. Beyond that I prefer to know what case it's appropriate to use this command. In my opinion, for example, we shouldn't exclude forces between O and O in bulk water system. However, can we exclude forces between C and C in graphene confined water system? Because the carbon atoms are fixed in the AIMD simulation.

**Reply by njzjz** [reply]
How do you calculate energy and forces using ab initio methods? The data should be consistent with your model.

**Reply by zhao-w-en** [reply]
We use VASP and the training data are generated from NVT simulations. In the simulation we fixed C atoms in positon. And we 
 consider all elements when we training a dp model. Is it reasonable to exclude carbon interactions on this basis?

**Reply by njzjz** [reply]
It's not related to simulations. If you exclude in the training, you should also exclude in first-principle calculation.

**Reply by njzjz** [reply]
If it's hard to do so in first-principle calculation, it's better not to do so...

**Reply by zhao-w-en** [reply]
Your advices are really helpful. It seems that we have to consider that interaction in training process. Besides, is there any  cases of training parameters and model accuracy about dp model trainning on github? As the amount of data becomes larger, the network seems harder to fit them. Very thanksful for your kind reply.

**Reply by njzjz** [reply]
https://docs.deepmodeling.org/projects/deepmd/en/latest/troubleshooting/howtoset_netsize.html

**Comment by njzjz** [comment]
@amcadmus Please help me transfer this discussion to deepmd-kit repo
